modules/compiler/spirv-ll/test/spvasm/debug_info.spvasm

```diff
@@ -95,11 +116,20 @@
 ; CHECK-SAME: {{(, unit: ![0-9]+)?}}
 ; CHECK-SAME: {{(, (variables|retainedNodes): ![0-9]+)?}}
 ; CHECK-SAME: )
-; CHECK: [[mainSubroutineType]] = !DISubroutineType(types: !{{[0-9]+}})
-; CHECK: [[aLocation]] = !DILocation(line: 6, column: 3, scope: [[mainLexicalBlock:![0-9]+]])
-; CHECK: [[mainLexicalBlock]] = distinct !DILexicalBlock(scope: [[mainSubprogram]], file: [[File]], line: 5, column: 2)
-; CHECK: [[bLocation]] = !DILocation(line: 7, column: 3, scope: [[mainLexicalBlock]])
-; CHECK: [[ifConditionLocation]] = !DILocation(line: 8, column: 3, scope: [[mainLexicalBlock]])
-; CHECK: [[ifTrueBlockLocation]] = !DILocation(line: 8, column: 3, scope: [[ifLexicalBlock:![0-9]+]])
-; CHECK: [[ifLexicalBlock]] = distinct !DILexicalBlock(scope: [[mainSubprogram]], file: !{{[0-9]+}}, line: 8, column: 3)
-; CHECK: [[ifJoinLocation]] = !DILocation(line: 9, column: 5, scope: [[ifLexicalBlock]])
+; CHECK-DAG: [[mainSubroutineType]] = !DISubroutineType(types: !{{[0-9]+}})
+; CHECK-DAG: [[aLocation]] = !DILocation(line: 6, column: 3, scope: [[mainLexicalBlock:![0-9]+]])
+; CHECK-DAG: [[mainLexicalBlock]] = distinct !DILexicalBlock(scope: [[mainSubprogram]], file: [[File]], line: 5, column: 2)
+; CHECK-DAG: [[bLocation]] = !DILocation(line: 7, column: 3, scope: [[mainLexicalBlock]])
+; CHECK-DAG: [[ifConditionLocation]] = !DILocation(line: 8, column: 3, scope: [[mainLexicalBlock]])
+; CHECK-DAG: [[ifTrueBlockLocation]] = !DILocation(line: 8, column: 3, scope: [[ifLexicalBlock:![0-9]+]])
+; CHECK-DAG: [[ifLexicalBlock]] = distinct !DILexicalBlock(scope: [[mainSubprogram]], file: !{{[0-9]+}}, line: 8, column: 3)
+; CHECK-DAG: [[ifJoinLocation]] = !DILocation(line: 9, column: 5, scope: [[ifLexicalBlock]])
+
+; CHECK: [[testfn2Subprogram]] = distinct !DISubprogram(name: "testfn2", linkageName: "testfn2", scope: null, file: [[File]],
+; CHECK-SAME: line: 5, type: [[mainSubroutineType]]
+; CHECK-SAME: {{(, isLocal: true, isDefinition: true)?}}, scopeLine: 1
+; CHECK-SAME: {{(, isOptimized: false)?}}
+; CHECK-SAME: {{(, spFlags: DISPFlagDefinition)?}}
+; CHECK-SAME: {{(, unit: ![0-9]+)?}}
+; CHECK-SAME: {{(, (variables|retainedNodes): ![0-9]+)?}}
```

Comment:
These `CHECK-SAME`s will always pass, regardless of what is output, because they match an empty string.

Reply:
Good point! I've updated them now. It's tricky to match against debug info because it changes so frequently. In this test, LLVM 16 puts `retainedNodes` leading to an empty list, but LLVM 17 omits it.